### DISCUSSION TITLE
stipple log file?

### DISCUSSION THREAD

**Discussion by ghost** [GenieFramework/Stipple.jl#139] [Q&A] [answered]
Hi all
new to all of this but LOVING it!!!   Knocking down my ignorance barriers one by one in this marvelous environment.  It's a real boon to a noob to see the results appear on a webpage. Very encouraging thanks to team stipple and genie.

One question. I am running my stipple script in cron and it worked well yesterday BUT something happened this morning. I can't find anything about a stipple log file to see what happened. I am using linux mint. There's nothing in the /var/log/syslog except the successful cron jobs.  but when I try 127.0.0.1:8000 there's nothing there and I can't see where it failed. Is there a log or should I add logging to my code.

**Comment by essenciary** [comment]
Thanks for the feedback :) 

You can control many aspects of the logging with `Genie.config.log_*` - ex: 
* `Genie.config.log_level`
* `Genie.config.log_to_file`

and `Genie.config.path_log`. 

Depending on how your app is set up, most likely logging to file is disabled. So you'll need to enable it. After you apply the changes restart the app. 

See here: https://github.com/GenieFramework/Genie.jl/blob/23659f356820d7af80b14360def5f4ecd2a45e7f/src/Configuration.jl#L164

**Reply by ghost** [reply]
thank you for such a wonderful package AND the reply.  I am not sure where I implement these changes as I am using Stipple/StippleUi/StipplePloty. Do I set these in my stipple script and they are passed to Genie?
thanks again

**Comment by essenciary** [comment, marked as answer]
Yes - set them early, for example as the first thing in the `routes.jl` file or in your script, depending on how your app is structured.

**Reply by ghost** [reply]
Hi there my code is structured below.  
```
using Stipple
using StippleUI
using StipplePlotly


@reactive mutable struct TontineModel <: ReactiveModel
    tontine_data::R{DataTable} = DataTable(df_table[])
end

function ui(model::TontineModel)
    page(
        model, class="container", title="title TONTINE2 ", head_content=Genie.Assets.favicon_support(),

        [
        heading("heading Tontine2 7/18/22")

        row([
            cell(class="st-module", [
            h5("h5 tontine data")
            table(:tontine_data)
            ])
        ])
        ]
    )
end

function handlers(model)
    on(df_table) do _
        notify(model.tontine_data)
    end

    model
end

route("/") do
    TontineModel |> init |> handlers |> ui |> html
end

# up(9000; async = true, server = Stipple.bootstrap())
up()
```
